.github/workflows/haskell.yml

```diff
@@ -45,9 +44,8 @@ jobs:
           path: |
             ~/.stack
             .stack-work
-          key: stack-${{ runner.os }}-${{ hashFiles('stack.yaml.lock') }}
+          key: stack-${{ runner.os }}-${{ hashFiles('stack.yaml.lock') }}-2
           restore-keys: |
-            stack-${{ runner.os }}
```

Comment:
Does this not disable caching altogether?

Reply:
When there is a cache whose key matches exactly, that cache is used.

Reply:
Ah, I think this part of the PR should most likely be kept then, since I am not sure what clearing the caches might achieve if it continues to be broken. I am also slightly surprised/concerned that the caches are causing build breakage: Stack should be able to deal with this stuff properly!